conventions/use-prettier/src/prettier-action.ts

```diff
@@ -0,0 +1,119 @@
+import { execSync } from 'child_process'
+import * as path from 'path'
+import {
+  CommandLineAction,
+  CommandLineStringParameter,
+} from '@rushstack/ts-command-line'
+import { FileSystem } from '@rushstack/node-core-library'
+import { resolveConfigFile } from 'prettier'
+
+import { resolveAndNormalizePath } from './utils'
+import type { BasePathPrefixChoices } from './types'
+
+/**
+ * Prettier wrapper
+ *
+ * Bookmarks:
+ * - https://github.com/microsoft/rushstack/blob/%40rushstack/ts-command-line_v4.4.6/apps/api-extractor/src/cli/InitAction.ts
+ */
+export class PrettierAction extends CommandLineAction {
+  private _configFileParameter!: CommandLineStringParameter
+
+  /**
+   * Fully qualified path to this package's provided prettier configuration.
+   */
+  public readonly CONFIG_FILE_PATH: string
+
+  /**
+   * Fully qualified path to this package's prettier binary.
+   */
+  public readonly BINARY_FILE_PATH: string
+
+  public constructor() {
+    super({
+      actionName: 'prettier',
+      summary: `Runs prettier`,
+      documentation: `Use this command to run prettier with shipped re-usable prettier config.`,
+    })
+    this.CONFIG_FILE_PATH = resolveAndNormalizePath('../prettier.config.js')
+    this.BINARY_FILE_PATH = resolveAndNormalizePath(
+      '../node_modules/.bin/prettier',
+    )
+  }
+
+  /**
+   * Attempt at resolving the syntax fixer configuration file.
+   *
+   * @param path Path to try loading from
+   * @param prefixWith Do we want as a local path, or from local folder
+   */
+  protected async resolveConfigFile(
+    path: string,
+    prefixWith?: BasePathPrefixChoices,
+  ): Promise<null | string> {
+    let usablePath: string = path
+    try {
+      const attempt = resolveAndNormalizePath(path, prefixWith)
+      usablePath = attempt
+    } catch (_) {
+      // Fall back to whatever given
+    }
+    return resolveConfigFile(usablePath)
+  }
+
+  protected onDefineParameters(): void {
+    /**
+     * Related to [@rushstack/ts-command-line][rushstack-1336];
+     * How can we merge this value from provided from CLI or
+     * serve our own.
+     *
+     * [rushstack-1336]: https://github.com/microsoft/rushstack/issues/1336
```

Comment:
Part related to reproducing or attempting how to use **defineCommandLineRemainder**